presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java

```diff
@@ -138,8 +138,20 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain)
                 }
             }
 
-            String min = minColumn(handle.getColumnId());
-            String max = maxColumn(handle.getColumnId());
+            String min;
+            String max;
+            if (handle.isBucketNumber()) {
+                if (!bucketed) {
+                    predicate.add("true");
```

Comment:
No, it should be true because we can have a query which says where `bucket_number = NULL` and we want to return rows for that case. In other cases the query will simply be slightly inefficient.